### DISCUSSION TITLE
Using psycopg over psycopg2

### DISCUSSION THREAD

**Discussion by vshah11** [sqlalchemy/sqlalchemy#11648] [Usage Questions] [answered]
I have sqlalchemy [2.0.31] and psycopg [3.~]
I wish to migrate from psycopg2 to psycopg (3) but ever time I call the sqlalchemy create_engine it keeps giving me 'module not found "psycopg2"'. I also updated the driver to reflect - postgresql+psycopg.

Any suggestions to resolve this would be helpful.

**Comment by CaselIT** [comment, marked as answer]
hi,

You seem to be using the wrong connection url. ensure that you are using something like `postgresql+psycopg://scott:tiger@127.0.0.1:5432/test`.

If you are can you post an example that reproduce it and the complete stack trace?

**Comment by vshah11** [comment]
Your a saviour. I did update my connection string with that drivername, however my database creation url still had "postgresql" as opposed to "postgresql+psycopg" which was causing the issue.
